### PR TITLE
Toning down infections

### PR DESCRIPTION
**Pull request by QualityVan**
## About The Pull Request
Tuning for wound/limb infection.
Wounds gaining germs is unchanged, but once they're closed they'll slowly clear out their own infection if it's not too high. Kitting helps here.
Slightly rescaled how fast wound germs transfer to the limb: a minimum amount are required so a fresh wound won't start transferring instantly.
Infected limbs are now much more apparent to their player, with repeating messages the same way broken bones show up. These're suppressed if you have spaceacillin in you.
Health analyzers have a lower threshold for detecting infected wounds.

Also a minor fix with wounds not changing their current state correctly when they worsen. Shouldn't affect function at all, just how they show up on examine.

## Why It's Good For The Game
They're a bit excessive right now. This might be too much, but there's plenty of knobs to tune back up later.

## Changelog
:cl:
qol: Infected limbs are less subtle to their owner.
balance: Wounds with smaller amounts of germs will now disinfect passively over time. Treatment speeds this up.
balance: Fresh wounds won't immediately start infecting the limb they're on.
fix: Wounds that worsen now do so more visibly.
/:cl:

### CODE REVIEW

File: code/modules/organs/limbs.dm [hunk 1]
```diff
@@ -452,17 +453,32 @@ Note that amputating the affected organ does in fact remove the infection from t
...
-			//Infected wounds raise the organ's germ level.
-			if (W.germ_level)
-				germ_level += max(1,round(W.germ_level/35))
-		else if (W.germ_level && prob(80)) //Antibiotics wont be very effective it the wound is still open
-			germ_level++
+		//Open wounds can become infected - spaceacillin won't help here.
+		if (owner.germ_level > W.germ_level && infection_checked && W.damage >= 20)
+			W.germ_level++
+
+		if(!W.germ_level)
```
Comment (BraveMole) [inline code/modules/organs/limbs.dm:466]:
```suggestion
		if(W.germ_level <= 0)
```

File: code/modules/organs/limbs.dm [hunk 2]
```diff
@@ -66,7 +66,8 @@
 	var/vital = FALSE
 	///INTERNAL germs inside the organ, this is BAD if it's greater than INFECTION_LEVEL_ONE
 	var/germ_level = 0
-
+	///Keeps track of the last time the limb bothered its owner about infection to prevent spam.
+	var/next_infection_message
```
Comment (BraveMole) [inline code/modules/organs/limbs.dm:70]: We have cooldown declaration that are doing just that

File: code/modules/organs/limbs.dm [hunk 3]
```diff
@@ -479,11 +495,12 @@ Note that amputating the affected organ does in fact remove the infection from t
 		if(prob(round(germ_level/10)))
 			if (spaceacillin < MIN_ANTIBIOTICS)
 				germ_level++
+				if (world.time > next_infection_message && (germ_level <= INFECTION_LEVEL_TWO) && !(limb_status & LIMB_NECROTIZED))
+					to_chat(owner, span_notice("Your [display_name] itches and feels warm..."))
+					next_infection_message = world.time + 60 SECONDS + rand(0, 300) //Next message 60-90 seconds from now.
```
Comment (BraveMole) [inline code/modules/organs/limbs.dm:500]:
```suggestion
					next_infection_message = world.time + rand(60 SECONDS, 90 SECONDS)//Next message 60-90 seconds from now.
```
But use cooldown start anyway

File: code/modules/organs/limbs.dm [hunk 4]
```diff
@@ -496,8 +513,9 @@ Note that amputating the affected organ does in fact remove the infection from t
 		if(prob(round(germ_level/10)))
 			if (spaceacillin < MIN_ANTIBIOTICS)
 				germ_level++
-		if (prob(1) && (germ_level <= INFECTION_LEVEL_THREE))
-			to_chat(owner, span_notice("Your infected wound itches and badly hurts!"))
+				if (world.time > next_infection_message && (germ_level <= INFECTION_LEVEL_THREE) && !(limb_status & LIMB_NECROTIZED))
+					to_chat(owner, span_warning("Your infected [display_name] is turning off-color and stings like hell!"))
+					next_infection_message = world.time + 25 SECONDS + rand(0, 150) //Next message 25-40 seconds from now.
```
Comment (BraveMole) [inline code/modules/organs/limbs.dm:518]: ditto

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/modules/organs/limbs.dm (hunk 2 above)
Comment (QualityVan) [inline code/modules/organs/limbs.dm:70]:
```suggestion
	COOLDOWN_DECLARE(next_infection_message)
``` Thusly?

**Review by BraveMole — APPROVED** [review]

**Review by BraveMole — APPROVED** [review]

**Pull request merged**